### ISSUE TITLE
hlint-3.0 regression: Unrecognized language extension 'Trustworthy'

### ISSUE THREAD

**Issue by RyanGlScott** [ndmitchell/hlint#961]
Consider the following program:

```hs
{-# LANGUAGE Trustworthy #-}
module Bug where
```

On `hlint-2.2.11`, this produces no hints, as expected:

```
$ hlint --version
HLint v2.2.11, (C) Neil Mitchell 2006-2020
$ hlint Bug.hs
No hints
```

With `hlint-3.0`, however, it crashes:

```
$ hlint --version
HLint v3.0, (C) Neil Mitchell 2006-2020
$ hlint Bug.hs
hlint: IMPOSSIBLE: Unrecognized language extension 'Trustworthy'
CallStack (from HasCallStack):
  error, called at src/Hint/Extensions.hs:266:11 in hlint-3.0-b26bf2d7d97a0573cc460418b91c96800f45ab73c13e16368e3465671e6a15bf:Hint.Extensions
```

**Comment by RyanGlScott** [comment]
`hlint-3.0` crashes similarly if you enable `Safe` or `Unsafe`.

**Comment by ndmitchell** [comment]
Eek. Thanks for the report. Looking into it now. Seems to be caused because Trustworthy doesn't have an entry in the GHC extensions data type: https://hackage.haskell.org/package/ghc-lib-parser-8.10.1.20200412/docs/GHC-LanguageExtensions-Type.html#t:Extension. CC @shayne-fletcher

**Comment by RyanGlScott** [comment]
The `Haskell98` and `Haskell2010` extensions also trigger the same crash.

**Comment by ndmitchell** [comment]
Thanks - I should have a fix within about 15 mins.

**Comment by ndmitchell** [comment]
Fix pushed. I'll wait for CI to kick in (15 mins or so) then make a release of HLint 3.0.1. Problem is that these are language pragmas that don't have an associated extension, which we assumed could never happen. I now handle that case in the extension hint, and have added test cases for it.

**Comment by shayne-fletcher** [comment]
Yes, they aren't extensions and thus don't appear in `xFlags`. It turns out there is a dedicated field in `DynFlags` of type `SafeHaskellMode`.
```
-- | The various Safe Haskell modes
data SafeHaskellMode
   = Sf_None          -- ^ inferred unsafe
   | Sf_Unsafe        -- ^ declared and checked
   | Sf_Trustworthy   -- ^ declared and checked
   | Sf_Safe          -- ^ declared and checked
   | Sf_SafeInferred  -- ^ inferred as safe
   | Sf_Ignore        -- ^ @-fno-safe-haskell@ state
   deriving (Eq)
```

**Comment by ndmitchell** [comment]
Released as 3.0.1. Thanks for the bug report!
